### ISSUE TITLE
Nvda does not read the last word of a text in java application

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#9997]
### Steps to reproduce:
Open a java application where there is a text editing field.
Enter a few words.
Press ctrl + end to go to the end of the text.
Navigate with ctrl + left arrow, and then move forward with ctrl + right arrow.
Note that the last word of the typed text is not read by nvda, it is read in white.
Navigation letter by letter or line by line pronounces the last word.
### Actual behavior:
when navigating word for word the last word is not read.
### Expected behavior:
The last word should be read when navigating word by word.
#### NVDA installed/portable/running from source:
portable
#### NVDA version:
alfa 18199
#### Windows version:
10 17763.615
#### Name and version of other software in use when reproducing the issue:
aplication java
#### Does the issue still occur after restarting your PC?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no
log:
[javaedit.txt](https://github.com/nvaccess/nvda/files/3446659/javaedit.txt)

**Comment by leonardder** [comment]
Could you test this with a Beta version of NVDA?

**Comment by fernando-jose-silva** [comment]
tested with the last alpha summer
18222, be63d19d
The failure has occurred.

I tested with the last rc:
nvda_snapshot_rc-18227, bc11064f.exe
The failure did not occur.

tested with the last beta
nvda_2019.2rc1.exe
18228
The failure did not occur either.

Thank you very much.

**Comment by DrSooom** [comment]
@fernando-jose-silva: Alpha-18178 (2019-07-25) and higher run on Python 3, the current Betas and RCs for NVDA 2019.2 on Python 2. Therefore this issue isn't fixed yet (for NVDA 2019.3). (I guess this is the case; I didn't test it.) @leonardder: Please correct me if I'm wrong.

**Comment by leonardder** [comment]
I think I would need access to an application that exposes this behavior to debug this.

**Comment by lukaszgo1** [comment]
From my tests even JAVA control panel exhibits this behavior when clicking on about, and reading the second edit field word by word.

**Comment by leonardder** [comment]
I'm sorry I missed this, I added this to the python 3 project, as this is really a regression. I'm diving into this straight ahead.
